bundles/org.openhab.binding.bluetooth.bluegiga/src/main/java/org/openhab/binding/bluetooth/bluegiga/BlueGigaBluetoothDevice.java

```diff
@@ -229,27 +245,20 @@ public void bluegigaEventReceived(BlueGigaResponse event) {
             }
 
             if (connectionState == ConnectionState.DISCOVERING) {
-                // We want to wait for an advertisement and a scan response before we call this discovered.
-                // The intention is to gather a reasonable amount of data about the device given devices send
-                // different data in different packets...
-                // Note that this is possible a bit arbitrary and may be refined later.
-                scanResponses.add(scanEvent.getPacketType());
-
-                if ((scanResponses.contains(ScanResponseType.CONNECTABLE_ADVERTISEMENT)
-                        || scanResponses.contains(ScanResponseType.DISCOVERABLE_ADVERTISEMENT)
-                        || scanResponses.contains(ScanResponseType.NON_CONNECTABLE_ADVERTISEMENT))
-                        && scanResponses.contains(ScanResponseType.SCAN_RESPONSE)) {
```

Comment:
Are we sure that all needed data is always received by one scan event or could it possible that bluetooth device send them as a separated scan events?